plugins/pur/core/src/main/java/com/pentaho/repository/importexport/StreamToJobNodeConverter.java

```diff
@@ -146,10 +147,18 @@ Repository connectToRepository() throws KettleException {
    */
   public IRepositoryFileData convert( final InputStream inputStream, final String charset, final String mimeType ) {
     try {
+      InputStream newInputStream = inputStream;
       long size = inputStream.available();
+
+      if ( size == 0 ) {
+        byte[] fileContent = IOUtils.toByteArray( newInputStream );
```

Comment:
commons-io has a CountingInputStream wrapper that could give a count at the end without having to hold everything in memory

Reply:
Thank you for your feedback! CountingInputStream is deprecated, so I used BoundedInputStream